src/main/java/uk/gov/hmcts/reform/userprofileapi/domain/entities/UserProfile.java

```diff
@@ -117,6 +118,9 @@ public void setStatus(IdamStatus status) {
     }
 
     public void setRoles(IdamRolesInfo idamrolesInfo) {
-        this.roles = idamrolesInfo.getRoles();
+        Optional.ofNullable(idamrolesInfo).ifPresent(
```

Comment:
Why do we need to set again () -> this.roles = new ArrayList<>()), because it is already initialised.
List<String> roles = new ArrayList<>();